giskard/llm/loaders/prompt_injections.py

```diff
@@ -0,0 +1,97 @@
+
+
+def _check_matching_dfs_len(df1, df2):
+    if len(df1) != len(df2):
+        raise ValueError(
+            f"{__name__}: {INJECTION_DATA_URL} and {GISKARD_META_URL} should "
+            "have the same length and should be a one-to-one mapping of each other."
+        )
+
+
+def _check_meta_df_requirements(df):
+    if "expected_strings" not in df.columns:
+        raise ValueError(f"{__name__}: expected_strings are needed for the evaluation.")
+
+    if df.expected_strings.isnull().values.any():
+        raise ValueError(f"{__name__}: expected_strings column cannot have any NaN values.")
+    df.expected_strings = df.expected_strings.apply(ast.literal_eval)
+
+
+class PromptInjectionDataLoader:
+    def __init__(
+        self,
+        num_samples: Optional[int] = None,
+    ):
+        self.num_samples = num_samples
+        self._df = None
+
+    def load_dataset_from_group(self, features, group) -> Dataset:
+        prompts = self.prompts_from_group(group)
+        prompts = pd.DataFrame({feature: prompts for feature in features}, index=prompts.index)
+        return Dataset(
+            df=prompts,
+            name="Injection Prompts",
+            target=None,
+            cat_columns=None,
+            validation=False,
+        )
+
+    @property
+    def df(self):
+        if self._df is None:
+            prompt_injections_df = pd.read_csv(INJECTION_DATA_URL, index_col=["index"])
+            meta_df = pd.read_csv(GISKARD_META_URL, index_col=["index"])
+            _check_matching_dfs_len(meta_df, prompt_injections_df)
+            _check_meta_df_requirements(meta_df)
+            self._df = prompt_injections_df.join(meta_df)
+
+            if self.num_samples is not None:
+                self._df = self._df.sample(self.num_samples)
+                self._df.reset_index(inplace=True, drop=True)
+
+        return self._df
+
+    @property
+    def names(self):
+        return self.df.name.tolist()
+
+    @property
+    def groups(self):
+        return list(set(self.df.group_mapping.tolist()))
+
+    def df_from_group(self, group):
+        return self.df.loc[self.df["group_mapping"] == group]
+
+    def prompts_from_group(self, group):
+        return self.df_from_group(group).prompt
+
+    def configs_from_group(self, group):
+        configs_df = self.df_from_group(group).drop(["prompt"], axis=1).to_dict("records")
+        configs = []
+        for row in configs_df:
+            kwargs = {k: v for k, v in row.items() if k in list(StringMatcherConfig.__annotations__.keys())}
+            configs.append(StringMatcherConfig(**kwargs))
+        return configs
+
+    def group_description(self, group):
+        group_description = self.df_from_group(group).description.to_list()
+        if len(set(group_description)) != 1:
+            raise ValueError(f"{self.__class__.__name__}: There must be only one group description per group.")
```

Comment:
Fault tolerance with warning instead of ValueError if multiple group_descriptions?
I would skip the check altogether and just return `group_description[0]`. If not present it will fail.

Data formatting checks should belong to the prompt-injection repo.

Reply:
done, I moved the validation to https://github.com/Giskard-AI/prompt-injections/blob/main/validate.py